lib/metasurfaces/Niyati_create_phasemap.m

```diff
@@ -0,0 +1,169 @@
+
+close all
+clear all
+
+%design = "vortex"; %
+design="roddier";
+trans  = "var"; %"const";
+lams   = [3.4,3.6,3.8,4.,4.2];
+% i_lam  = 2;%2;%3;%4;%5
+vortcube = [];%zeros(4);
+
+
+% mask = exp(phaseScaleFac*1j*roddiervort);
+
+
+for i_lam = 1:length(lams)
+
+    Npad   = 4096; %4096; %8192; % %2^13;
+    charge = 6;
+    theta  = angle(falco_gen_vortex_mask(1,Npad)); % gen_v_mask  with charge 1 means having an array with the azi angle for each pxl
+    z1 = 2.40483; % wiki Bessel zeros
+    z2 = 5.52008; % using z2 or z3 results in more than 2pi phase coverage needed
+    z3 = 8.65373; %
+    
+    %create roddier+sawtooth phase ideal pattern
+    coords = generateCoordinates(Npad);% Creates NxN arrays with coordinates 
+    roddiervort = 0.* coords.THETA;
+    domain = (coords.THETA >= 0);
+    roddiervort(domain) = charge*rem(coords.THETA(domain),2*pi./charge);
+    domain = (coords.THETA >= -pi) & (coords.THETA < 0);
+    roddiervort(domain) = charge*rem((coords.THETA(domain)+pi),2*pi./charge);
+    res = 1024;
+    R1 = (coords.RHO <= 0.53*res);
+    roddiervort(R1) =roddiervort(R1) + 0.5*2*pi;
+    
+    %TRANSLATE TO METASURFACE MAP
+    if design=="vortex", [pss,Tss,sss] = b30_library_siz_auto(angle(exp(1i.*charge.*theta)),i_lam); end % VORTEX: add wavelength dependent phase. 
+    if design=="roddier", [pss,Tss,sss] = b30_library_siz_auto(angle(exp(1i.*roddiervort)),i_lam); end % RODDIER: add wavelength dependent phase. 
+    if design=="cosine", [pss,Tss,sss] = b30_library_siz_auto(angle(exp(1i.*z1.*cos(charge.*theta)))+z1-pi,i_lam); end % COSINE: add wavelength dependent phase. 
+    vortex1 = exp(1i.*pss); % phase only
+    if trans=="const", vortex = vortex1; end
+    if trans=="var", vortex = vortex1.*sqrt(Tss); end % add phase dependent transmission values
+    
+    vortcube = cat(3,vortcube,vortex); %(:,:,i_lam);
+    
+end
+
+
+%% 1D profile
+
+%%% for (sawtooth) MS vortex
+lp=6; % topological charge
+N=200 % azimuthal sampling
+[phase,transmission,size]=b30_library_siz_auto(angle(exp(1i.*linspace(0,lp*2*pi,N))),linspace(1,5,5));
+i_lam=5; % wavelength index 3 is "perfect" (design wavelength)
+offsetphase = phase(:,i_lam);
+figure()
+plot(offsetphase,'o-')
+figure()
+plot(phase(:,i_lam)-phase(:,3),'o-')
+
+%%Fourier Transform:
+fftmetasurface = abs(fftshift(fft(offsetphase)))/N;
+
+%%Frequency specifications:
+Fs = N;
+dF = Fs/N;  %1                  % hertz
+f = -Fs/2:dF:Fs/2-dF;           % hertz
+
+%%Plot the spectrum:
+figure(311);
+hold on
+% plot(f,fftmetasurface);
+stem(f,fftmetasurface,'d','MarkerSize',10,'LineWidth',2,'Color',[0.9290, 0.6940, 0.1250]);
+xlim([-2 20]);
+ylim([1E-5 10])
+xticks([0:4:18])
+set(gca, 'YScale', 'log')
+ylabel('|C_m|^2'); 
+xlabel('Mode');
+title("FFT of Metasurface Phase Profile")
+
+
+
+%%
+save lorenzoroddierattempt.mat vortcube lams design
+% save lorenzovortexmap3.mat vortex
+
+
+
+%% Read in Skyler Metasurface Maps
+
+
+SkylerMapsDir = '/Users/niyatid/falco-matlab/metasurfaces/';
```

Comment:
You can replace this with `SkylerMapsDir = fileparts(mfilename('fullpath'))` to work for everybody assuming all the .txt and .mat files needed as inputs are included in this same folder.

Reply:
removed this script altogether--it's not ready for broader falco users anyways.